header/p2p/response.go

```diff
@@ -0,0 +1,33 @@
+package p2p
+
+import (
+	"fmt"
+)
+
+// ExtendedHeaderResponse is a Message used as a reply to a call ExtendedHeaderRequest.
+type ExtendedHeaderResponse struct {
+	// Body is a header representation in bytes.
+	Body []byte
+	// StatusCode contains the state of the ExtendedHeaderRequest(success/not success).
+	StatusCode uint32
+}
+
+// MarshalBinary marshals ExtendedHeaderResponse to binary.
+func (ehr *ExtendedHeaderResponse) MarshalBinary() ([]byte, error) {
+	return MarshalExtendedHeaderResponse(ehr)
+}
+
+// UnmarshalBinary converts binary to the ExtendedHeaderResponse.
+func (ehr *ExtendedHeaderResponse) UnmarshalBinary(data []byte) error {
+	if ehr == nil {
+		return fmt.Errorf("header: cannot UnmarshalBinary - nil ExtendedHeaderResponse")
+	}
+
+	out, err := UnmarshalExtendedHeaderResponse(data)
+	if err != nil {
+		return err
+	}
+
+	*ehr = *out
+	return nil
+}
```

Comment:
I feel like we don't need this. Nor imported publicly nor privately. I know that we do that for the Request, but we should not. It doesn't give us anything besides additional allocations, so can you pls make it work directly with PB structs and stop using Request definition in the subsequent issue/PR

Reply:
Yes, you are right. It was my first thought, but then I decided to make it similar to `request`.

Reply:
Pls, let's add to the epic the removal of Request structure

Reply:
https://github.com/celestiaorg/celestia-node/pull/1141